File: Strings/pagina06/3313/3313.php
<?php

function wordPlay($palavra, $i){
    $menor = $palavra;
    $maior = $palavra;
    $dupla = $palavra + $palavra;

    for ($j = 1; $j < strlen($palavra); $j++) {
        $rotacionada = substr($dupla, $j, strlen($palavra));

        if ($rotacionada < $menor) $menor = $rotacionada;
        if ($rotacionada > $maior) $maior = $rotacionada;
    }

    echo "Caso $i: $menor $maior\n";
}

$i = 1;
while(($palavra = readline()) != "*"){
    wordPlay($palavra, $i);
    $i++;
}

File: Strings/pagina06/3313/3313.cpp
#include <iostream>
#include <string>
using namespace std;

void wordplay(const string& palavra, int caso) {
    string menor = palavra;
    string maior = palavra;
    string dupla = palavra + palavra;

    for (int j = 1; j < palavra.size(); j++) {
        string rotacionada = dupla.substr(j, palavra.size());
        if (rotacionada < menor) menor = rotacionada;
        if (rotacionada > maior) maior = rotacionada;
    }

    cout << "Caso " << caso << ": " << menor << " " << maior << endl;
}

int main() {
    string palavra;
    int i = 1;
    while (cin >> palavra && palavra != "*") {
        wordplay(palavra, i);
        i++;
    }
    return 0;
}
